system/src/system_mode.cpp

```diff
@@ -105,3 +136,38 @@ spark::feature::State system_thread_get_state(void*)
 }
 
 #endif
+
+int system_reset(unsigned mode, unsigned reason, unsigned value, unsigned flags, void* reserved) {
+    unsigned defaultReason = RESET_REASON_NONE;
+    switch (mode) {
+    case SYSTEM_RESET_MODE_NORMAL:
```

Comment:
It's certainly cleaner, but I am wondering if the overhead of mapping values is worth it, compared to simply using the reset reason as the reset mode.

Reply:
The idea was to clearly separate reset modes (i.e. _how_ we want to reset) and reset reasons (_why_ we want to reset), so that it is possible to perform a specific type of a reset, e.g. a DFU reset, and still use whichever reason code would be useful for diagnostic purposes along with it.